### ISSUE TITLE
False-positive `TCH003`: `lint.typing-modules` setting is not honored in the typing module itself

### ISSUE THREAD

**Issue by bersbersbers** [astral-sh/ruff#11733]
`lint.typing-modules` allows defining compatibility modules whose exports will be treated the same as the exports from the `typing` module. However, this does not seem to work in the module itself - `ruff` does not seem to understand that `TYPE_CHECKING` defines a type-checking block in this example:

`bug.py`
```python
from __future__ import annotations

import typing

def is_type_checking() -> bool:
    # This will be more complicated, obviously
    return 4 < 10

# This should be considered a typing module's TYPE_CHECKING variable:
TYPE_CHECKING = typing.TYPE_CHECKING or is_type_checking()

if TYPE_CHECKING:
    # So I consider this a false-negative TCH003
    from collections.abc import Iterable

def fun(_var: Iterable[str]) -> None:
    return None
```

`pyproject.toml`
```toml
[tool.ruff]
lint.select = ["TCH003"]
lint.typing-modules = ["bug"]
```

`ruff check bug.py`
```
bug.py:14:33: TCH003 Move standard library import `collections.abc.Iterable` into a type-checking block
Found 1 error.
No fixes available (1 hidden fix can be enabled with the `--unsafe-fixes` option).
```

`ruff --version`
```
ruff 0.4.7
```

**Comment by bersbersbers** [comment]
I think this can be closed as a subset/duplicate of the now-closed https://github.com/astral-sh/ruff/pull/15719.
